### PR TITLE
[Windows] Execute console_scripts in foreground

### LINKED ISSUE #4481: [Windows] [Issue]: hipcc.exe writes terminal output after process termination
### Problem Description

That is the only way I can think of to describe this behavior. It is not limited to `hipcc`, it appears to be all the executables in the `Scripts` directory.

Image

### Operating System

Windows 11

### CPU

N/A

### GPU

N/A

### ROCm Version

7.*

### ROCm Component

_No response_

### Steps to Reproduce

Run `hipcc --help` as shown in the screenshot above.

### (Optional for Linux users) Output of /opt/rocm/bin/rocminfo --support

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by astrelsky**
## Motivation

The `console_scripts` provided in the rocm python packages currently execute in the background on Windows.

## Technical Details

The current behavior is due to a well known bug in Python and it is unlikely to be fixed. https://github.com/python/cpython/issues/63323

The approach of using `os.spawnv(os.P_WAIT, ...)` paired with `os._exit` as seen in the issue linked above.

## Test Plan

Tested locally.

## Test Result

The rocm executable runs in the foreground and behaves as expected.

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

Fixes #4481

### CODE REVIEW

**Comment by astrelsky** [comment]
If possible it would be cool to avoid running the full build and test suites since this only makes a change in the python packaging. Save the resources for more important testing.

**Comment by ScottTodd** [comment]
Full CI run is fine here, mostly looking to run the python tests. The general resource savings will come when we finish implementing https://github.com/ROCm/TheRock/issues/3399. Until then, we _could_ manually test that by using explicit prebuilt artifacts + the ci:skip label on this PR, following the docs at https://github.com/ROCm/TheRock/blob/main/docs/development/ci_behavior_manipulation.md#prebuilt-stages-multi-arch-ci

**Comment by ScottTodd** [comment]
Eh actually fine, I'll trigger those test jobs,1 sec.

**Comment by ScottTodd** [comment]
Testing at https://github.com/ROCm/TheRock/actions/runs/24368105372 (should use prebuilt artifacts for all stages and then skip to building + testing python packages)

**Comment by astrelsky** [comment]
I only thought to mention the ci because windows builds still take a few hours.


I just realize my thinking was under the assumption that it was limited to one set of jobs at a time (one pr at a time), I don't know if this is the case.

**Comment by astrelsky** [comment]
> Testing at https://github.com/ROCm/TheRock/actions/runs/24368105372 (should use prebuilt artifacts for all stages and then skip to building + testing python packages)

Is that the broken windows machine :joy:

**Comment by ScottTodd** [comment]
Yeah lots of our test runners are having issues today :/

The python package build at least succeeded + `rocm-sdk test` passed. That's a good sign. I'm more wondering if there are edge cases with processes like `amdclang` that take multiple seconds/minutes to run, as this now blocks instead of returning immediately. The pytorch build should exercise those paths.

**Comment by astrelsky** [comment]
> Yeah lots of our test runners are having issues today :/
> 
> The python package build at least succeeded + `rocm-sdk test` passed. That's a good sign. I'm more wondering if there are edge cases with processes like `amdclang` that take multiple seconds/minutes to run, as this now blocks instead of returning immediately. The pytorch build should exercise those paths.

Well, it should have been blocking in the first place shouldn't it have been? I can't imaging what erratic behavior it could have been causing.

There _might_ be slightly more memory usage from this change though since the python process is still running.

**Comment by ScottTodd** [comment]
PyTorch build succeeded: https://github.com/ROCm/TheRock/actions/runs/24368105372/job/71168010051. I see the point about memory usage but I think it's fine. Build time is at least within the noise there.

I further tested this locally and I can confirm that the issue is fixed:

```bash
# Before
D:\scratch\therock
(3.12.venv) λ pip install --index-url https://rocm.nightlies.amd.com/v2/gfx110X-all/ --pre rocm[devel] --upgrade
Successfully installed rocm-7.13.0a20260413 rocm-sdk-core-7.13.0a20260413 rocm-sdk-devel-7.13.0a20260413

D:\scratch\therock
(3.12.venv) λ hipconfig --path

D:\scratch\therock
(3.12.venv) λ D:\scratch\therock\3.12.venv\Lib\site-packages\_rocm_sdk_devel
# Note: output is _after_ the next shell line appears
```

```bash
# After
D:\scratch\therock
λ py -V:3.12 -m venv 3.12_foreground.venv && .\3.12_foreground.venv\Scripts\activate.bat

D:\scratch\therock
(3.12_foreground.venv) λ pip install rocm[libraries,devel] --pre ^     --find-links=https://therock-ci-artifacts.s3.amazonaws.com/24368105372-windows/python/gfx110X-all/index.html

D:\scratch\therock
(3.12_foreground.venv) λ hipconfig --path
D:\scratch\therock\3.12_foreground.venv\Lib\site-packages\_rocm_sdk_devel
# Note: output is inline as expected, next shell line appears afterwards)

D:\scratch\therock
(3.12_foreground.venv) λ
```

Thanks for finding the cpython issue and this minimal fix :)

**Review by ScottTodd — APPROVED** [review]

File: build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py [hunk 1]
```diff
@@ -102,6 +102,10 @@ def _exec(relpath: str, expand_devel=True):
     # need the devel files. System info tools (amd-smi, rocminfo, etc.)
     # override with expand_devel=False to avoid the expansion cost.
     full_path = _get_module_path(expand_devel) / (relpath + exe_suffix)
+    if is_windows:
+        # https://bugs.python.org/issue19124
```
Comment (ScottTodd) [inline build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py:106]: Want to use the new (redirected) URL or keep this old URL?

```suggestion
        # https://github.com/python/cpython/issues/63323
```

**Review by ScottTodd — APPROVED** [review]

File: build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py (hunk 1 above)
Comment (astrelsky) [reply build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py:106]: I'll leave it up to you. I only found the GitHub issue when opening the pull request and used it for referencing. The old link demonstrates how long it's been a problem in python though. I going to sleep so I won't change it until the morning.

Comment (ScottTodd) [reply build_tools/packaging/python/templates/rocm-sdk-core/src/rocm_sdk_core/_cli.py:106]: Okay to keep as-is then. I'll merge this.
